pkg/scaler/scaler.go

```diff
@@ -424,13 +436,26 @@ func (s *Scaler) GC(ctx context.Context) {
 	if err := s.gc(ctx, logger); err != nil {
 		s.metrics.numFailedGC.WithLabelValues(s.backend.Name()).Inc()
 
+		s.mu.Lock()
+		defer s.mu.Unlock()
+		s.lastErr = time.Now()
+
 		logger.Error(err)
 	}
 }
 
 func (s *Scaler) gc(ctx context.Context, logger *log.Entry) error {
 	logger.Debug("starting GC")
 
+	lastErr := time.Since(s.lastErr)
+	lastErrPeriod := time.Duration(s.opt.ErrGracePeriod)
+
+	if lastErr < lastErrPeriod {
+		logger.Infof("still in last error period. skipping gc: %v < %v", lastErr, lastErrPeriod)
+
+		return nil
+	}
```

Comment:
instead of this logic, do this in the client itself, and block the request there, so you can block requests in signle place, which will benefit in every place this client is used, not only in `gc`